src/JNI.jl

```diff
@@ -0,0 +1,767 @@
+module JNI
+
+# jnienv.jl exports
+export JNINativeInterface, JNIEnv, JNIInvokeInterface, JavaVM
+# jni_md.h exports
+export jint, jlong, jbyte
+# jni.h exports
+export jboolean, jchar, jshort, jfloat, jdouble, jsize, jprimitive
+# constant export
+export JNI_TRUE, JNI_FALSE
+# Legacy exports
+export jnifunc
+
+include("jnienv.jl")
+
+global jnifunc
+
+# jni_md.h
+const jint = Cint
+#ifdef _LP64 /* 64-bit Solaris */
+# typedef long jlong;
+const jlong = Clonglong
+const jbyte = Cchar
+
+# jni.h
+
+const jboolean = Cuchar
+const jchar = Cushort
+const jshort = Cshort
+const jfloat = Cfloat
+const jdouble = Cdouble
+const jsize = jint
+jprimitive = Union{jboolean, jchar, jshort, jfloat, jdouble, jint, jlong}
+
+jobject = Ptr{Nothing}
+jclass = Ptr{Nothing}
+jthrowable = Ptr{Nothing}
+jweak = Ptr{Nothing}
+jmethodID = Ptr{Nothing}
+jfieldID = Ptr{Nothing}
+jstring = Ptr{Nothing}
+jarray = Ptr{Nothing}
+JNINativeMethod = Ptr{Nothing}
+jobjectArray = Ptr{Nothing}
+jbooleanArray = Ptr{Nothing}
+jbyteArray = Ptr{Nothing}
+jshortArray = Ptr{Nothing}
+jintArray = Ptr{Nothing}
+jlongArray = Ptr{Nothing}
+jfloatArray = Ptr{Nothing}
+jdoubleArray = Ptr{Nothing}
+jcharArray = Ptr{Nothing}
+jvalue = Int64
+
+@enum jobjectRefType begin
+    JNIInvalidRefType    = 0
+    JNILocalRefType      = 1
+    JNIGlobalRefType     = 2
+    JNIWeakGlobalRefType = 3
+end
+
+const JNI_VERSION_1_1 = convert(Cint, 0x00010001)
+const JNI_VERSION_1_2 = convert(Cint, 0x00010002)
+const JNI_VERSION_1_4 = convert(Cint, 0x00010004)
+const JNI_VERSION_1_6 = convert(Cint, 0x00010006)
+const JNI_VERSION_1_8 = convert(Cint, 0x00010008)
+const JNI_VERSION_9   = convert(Cint, 0x00090000)
+const JNI_VERSION_10  = convert(Cint, 0x000a0000)
+
+const JNI_TRUE = convert(Cchar, 1)
+const JNI_FALSE = convert(Cchar, 0)
+
+# Return Values
+const JNI_OK           = convert(Cint, 0)               #/* success */
+const JNI_ERR          = convert(Cint, -1)              #/* unknown error */
+const JNI_EDETACHED    = convert(Cint, -2)              #/* thread detached from the VM */
+const JNI_EVERSION     = convert(Cint, -3)              #/* JNI version error */
+const JNI_ENOMEM       = convert(Cint, -4)              #/* not enough memory */
+const JNI_EEXIST       = convert(Cint, -5)              #/* VM already created */
+const JNI_EINVAL       = convert(Cint, -6)              #/* invalid arguments */
+
+# There is likely over specification here
+PtrIsCopy = Union{Ptr{jboolean},Ref{jboolean},Array{jboolean,}}
+AnyString = Union{AbstractString,Cstring,Ptr{UInt8}}
+
+function load_jni(penv::Ptr{JNIEnv})
+    jnienv = unsafe_load(penv)
+    global jnifunc = unsafe_load(jnienv.JNINativeInterface_) #The JNI Function table
+end
+is_jni_loaded() = isdefined(JNI, :jnifunc) 
+
+# === Below Generated by make_jni2.jl ===
+
+#export GetVersion
```

Comment:
maybe get rid of the commented exports in the generated code?

Reply:
Done with c041f1d